openassessment/xblock/static/js/spec/lms/oa_datefactory.js

```diff
@@ -0,0 +1,84 @@
+/**
+Tests for OA dateutil factory.
+**/
+
+describe('OpenAssessment.DateTimeFactory', function() {
+
+    beforeEach(function() {
+        // Load the DOM fixture
+        loadFixtures('oa_response_date.html');
+
+    });
+    describe('transform', function() {
+        it('has the correct HTML elements', function() {
+            var timeElement = $('.step__title').get(0);
+            var datetimeFactory = new OpenAssessment.DateTimeFactory(timeElement);
+            datetimeFactory.report();
+            $('.ora-datetime', timeElement).each(function() {
```

Comment:
I'm nervous about this as ``report`` is asynchronous because it invokes ``generate`` which in turn uses ``require``. It seems that there might be a timing window here where the factory hasn't completed before these assertions are run.

Thinking it through, I think the three dependencies (jquery, string-utils, and date-utils) should all be pre-loaded both in platform and in the tests, so maybe in that case the require call is synchronous.